ci-runner/helper/DockerHelper.go

```diff
@@ -1054,6 +1057,70 @@ func getBuildXDriverOptionsWithImage(buildxDriverImage, driverOptions string) st
 	return driverOptions
 }
 
+func getBuildXDriverOptionsWithLabelsAndAnnotations(driverOptions string) string {
+	labels := make(map[string]string)
+	annotations := make(map[string]string)
+
+	// Read labels and annotations from files
+	labelsPath := utils.DEVTRON_SELF_DOWNWARD_API_VOLUME_PATH + "/labels"
+	labelsOut, err := os.ReadFile(labelsPath)
+	if err != nil {
+		log.Println(util.DEVTRON, "error in reading labels from podinfo, err:", err)
+	}
+	annotationsPath := utils.DEVTRON_SELF_DOWNWARD_API_VOLUME_PATH + "/annotations"
+	annotationsOut, err := os.ReadFile(annotationsPath)
+	if err != nil {
+		log.Println(util.DEVTRON, "error in reading annotations from podinfo, err:", err)
+	}
+
+	// Parse labels and annotations
+	if len(labelsOut) > 0 {
+		labels = parseKeyValuePairs(string(labelsOut))
+	}
+	if len(annotationsOut) > 0 {
+		annotations = parseKeyValuePairs(string(annotationsOut))
+	}
+
+	// Combine driver options
+	driverOptions = getBuildXDriverOptions("labels", labels, driverOptions)
+	driverOptions = getBuildXDriverOptions("annotations", annotations, driverOptions)
+	return driverOptions
+}
+
+func parseKeyValuePairs(input string) map[string]string {
+	keyValuePairs := make(map[string]string)
+	lines := strings.Split(input, "\n")
+	for _, line := range lines {
+		if len(line) > 0 {
+			kv := strings.SplitN(line, "=", 2)
+			if len(kv) == 2 {
+				key := kv[0]
+				value := strings.Trim(kv[1], `"`)
+				keyValuePairs[key] = value
+			}
+		}
+	}
+	return keyValuePairs
+}
+
+func getBuildXDriverOptions(optionType string, options map[string]string, driverOptions string) string {
```

Comment:
use library/utility function

Reply:
revisit this logic within buildx and explore library

Reply:
<img width="876" alt="Screenshot 2024-12-31 at 6 05 34 PM" src="https://git
<img width="892" alt="Screenshot 2024-12-31 at 6 05 46 PM" src="https://github.com/user-attachments/assets/12d541a3-f417-421b-96b5-3cc332d275de" />
hub.com/user-attachments/assets/d05eb13b-0790-464e-8a5c-ecd29c5e444f" />

Reply:
buildx has also written parsing logic in same manner